src/consumers/create-transformed-erc20-event-fills/index.js

```diff
@@ -0,0 +1,210 @@
+   */
+  const bridgeEvents = await Event.find({
+    transactionHash: transformedERC20Event.transactionHash,
+    type: 'ERC20BridgeTransfer',
+  }).lean();
+
+  /*
+   * If there are no ERC20BridgeTransfer then the transform would have occurred
+   * using traditional fills which will be handled through the standard workflow
+   */
+  if (bridgeEvents.length === 0) {
+    logger.info(
+      `TransformedERC20 event has no associated ERC20BridgeTransfer events: ${eventId}`,
+    );
+    return;
+  }
+
+  // const transformMakerAmount = new BigNumber(
+  //   transformedERC20Event.data.inputTokenAmount,
+  // );
+  // const transformTakerAmount = new BigNumber(
+  //   transformedERC20Event.data.outputTokenAmount,
+  // );
+
+  // const { bridgeMakerAmount, bridgeTakerAmount } = bridgeEvents.reduce(
+  //   (acc, current) => ({
+  //     bridgeMakerAmount: acc.bridgeMakerAmount.plus(
+  //       current.data.fromTokenAmount,
+  //     ),
+  //     bridgeTakerAmount: acc.bridgeTakerAmount.plus(current.data.toTokenAmount),
+  //   }),
+  //   {
+  //     bridgeMakerAmount: new BigNumber(0),
+  //     bridgeTakerAmount: new BigNumber(0),
+  //   },
+  // );
+
+  // TODO: Potentially this guard needs to be removed
+  // if (
+  //   bridgeMakerAmount.gt(transformMakerAmount) ||
+  //   bridgeTakerAmount.gt(transformTakerAmount)
+  // ) {
+  //   throw new Error(
+  //     `Transaction has TransformedERC20/ERC20BridgeTransfer mismatch: ${event.transactionHash}`,
+  //   );
+  // }
+
+  // TODO: Should we consider an additional guard which verifies TransformedERC20 tokens match
+  // the ERC20BridgeTransfer tokens? Will need special case for ETH.
+
+  /*
+   * If the job has made it this far then we're comfortable enough that fill documents can
+   * be created for the associated ERC20BridgeTransfer events. We assume that each ERC20BridgeTransfer
+   * event is related to the TransformedERC20 event being processed and will use the TransformedERC20
+   * event to dictate the token and taker addresses.
+   */
+  const fills = bridgeEvents.map(bridgeEvent => ({
+    _id: bridgeEvent._id,
+    affiliateAddress: transaction.affiliateAddress,
+    assets: [
+      {
+        actor: FILL_ACTOR.MAKER,
+        amount: new BigNumber(bridgeEvent.data.fromTokenAmount).toNumber(),
+        bridgeAddress: bridgeEvent.data.from,
+        tokenAddress: bridgeEvent.data.fromToken,
+      },
+      {
+        actor: FILL_ACTOR.TAKER,
+        amount: new BigNumber(bridgeEvent.data.toTokenAmount).toNumber(),
+        tokenAddress: bridgeEvent.data.toToken,
+      },
+    ],
+    blockHash: transaction.blockHash,
+    blockNumber: transaction.blockNumber,
+    date: transaction.date,
+    eventId: bridgeEvent._id,
+    logIndex: bridgeEvent.logIndex,
+    maker: bridgeEvent.data.from,
+    protocolVersion: bridgeEvent.protocolVersion,
+    quoteDate: transaction.quoteDate,
+    taker: transformedERC20Event.data.taker,
```

Comment:
This can be taken from the bridge. from the bridge event--'to'

Reply:
This looks like it wouldn't work in the case when `ERC20BridgeTransfer.to` is the 0x flash wallet though?

Take for example https://etherscan.io/tx/0x493d99ebbfd8dfa9cefba6fe291c22c728db036477fc5186fcb1518b667cbc36#eventlog which has `ERC20BridgeTransfer.to` as `0x22f9dcf4647084d6c31b2765f6910cd85c178c18` whereas `TransformedERC20.taker` is `0x3e880db87082197292e2a7447f2c39a3c55e0a8a`.